src/platform/plugins/private/presentation_panel/public/panel_component/panel_header/use_presentation_panel_header_actions.tsx

```diff
@@ -86,11 +86,20 @@ export const usePresentationPanelHeaderActions = <
       );
       if (canceled) return;
       for (const badge of frequentlyChangingBadges) {
-        subscriptions.add(
-          badge.subscribeToCompatibilityChanges(apiContext, (isCompatible, action) =>
-            handleActionCompatibilityChange('badge', isCompatible, action as AnyApiAction)
+        const compatibilitySubject = badge
+          .getCompatibilityChangesSubject(apiContext)
```

Comment:
I wonder if we should make a sugar method for these in the uiActions plugin or on the base Action definition class? That way consumers don't all have to set up the `switchMap`.

Reply:
Discussed offline. The issue is complicated by the fact that actions are defined with multiple types, ActionDefintion, Action, and ActionInternal. There is not a good place to put this since changing Action. getCompatibilityChangesSubject's type to `Observable<boolean>` would impact action location where actions are created using a class. The first step is to only allow actions to be registered with `ActionDefintion` instead of Action class.

Decided to leave as-is for now.